### PR TITLE
Display error page in case an unmatched route is hit

### PR DESCRIPTION
**Pull request by tmscarla**
## Description

Currently, pointing to a non-existing page in PCUI returns an empty page, which can be confusing for customers.

This PR introduces a new error page template in PCUI that displays a clear message to the user indicating that an error has occurred, along with options for resolving the issue.

## Changelog entry
* Add error page to be displayed if an unmatched route is hit

## How Has This Been Tested?
* Manual tests navigating to unmatched routes, even in the case where the frontend is served by the backend

Screenshot 2023-01-18 at 23 04 35

## PR Quality Checklist

- [ ] I added tests to new or existing code
- [ ] I removed hardcoded strings and used our `i18n` solution instead (see [here](https://github.com/aws-samples/pcluster-manager/pull/175/commits/fdc6b77987c87a26f51dbc8da5d371d95ef80601))
- [ ] I made sure no sensitive info gets logged at any time in the codebase (see [here](https://cheatsheetseries.owasp.org/cheatsheets/Logging_Cheat_Sheet.html)) (e.g. no user info or details, no stacktraces, etc.)
- [ ] I checked that infrastructure/update_infrastructure.sh runs without any error
- [x] I checked that `npm run build` builds without any error
- [x] I checked that clusters are listed correctly
- [ ] I checked that a new cluster can be created (config is produced and dry run passes)
- [ ] I checked that login and logout work as expected

In order to increase the likelihood of your contribution being accepted, please make sure you have read both the [Contributing Guidelines](../CONTRIBUTING.md) and the [Project Guidelines](../PROJECT_GUIDELINES.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: frontend/src/components/NoMatch.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import {Box, Link} from '@cloudscape-design/components'
+import React from 'react'
+import TopBar from './TopBar'
+import errorPage from './../../public/img/error_pages_illustration.svg'
```
Comment (mendaomn) [inline frontend/src/components/NoMatch.tsx:4]: issue: please make sure this illustration is covered by our licenses ([see for instance here](https://github.com/aws-samples/pcluster-manager/blob/main/frontend/public/flags/Flags.LICENSE))

File: frontend/src/components/NoMatch.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+              <Link
+                external
+                externalIconAriaLabel={t('global.openNewTab')}
+                href={t('global.docs.link')}
+              >
+                {t('global.docs.title')}
+              </Link>
+            </li>
+          </ul>
+        </Box>
```
Comment (mendaomn) [inline frontend/src/components/NoMatch.tsx:43]: issue: for consistency of typography, see if you can use components provided by the design system. For instance the [TextContent component](https://cloudscape.design/components/text-content/?tabId=playground) seems a good choice here

**Comment by mtfranchetto** [comment]
@tmscarla issue: have you tried testing the frontend when served from the backend? Serving from the backend as a public folder differs from going to Next.js directly. If you try accessing a non-existing path on Flask at the moment you received a 404 error, and the index.html is not served.

File: frontend/src/components/NoMatch.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import {useTranslation} from 'react-i18next'
+import {useLocationChangeLog} from '../navigation/useLocationChangeLog'
+
+export function NoMatch() {
+  const {t} = useTranslation()
+  useLocationChangeLog()
+
+  return (
+    <>
+      <TopBar />
```
Comment (mendaomn) [inline frontend/src/components/NoMatch.tsx:15]: issue: reuse our `Layout` component for consistency (I don't think there should be an issue with showing the navigation bar here)

Also, for better placement of the content try to build the page the Cloudscape way. For instance see if [Container](https://cloudscape.design/components/container/?tabId=playground) or [ContentLayout](https://cloudscape.design/components/content-layout/?tabId=playground) can be useful here

**Comment by mendaomn** [comment]
test: let's add an e2e for this case

File: frontend/src/components/NoMatch.tsx (hunk 2 above)
Comment (tmscarla) [reply frontend/src/components/NoMatch.tsx:43]: Right, thanks!

File: frontend/src/components/NoMatch.tsx (hunk 3 above)
Comment (tmscarla) [reply frontend/src/components/NoMatch.tsx:15]: Going to use `Layout` component

**Comment by tmscarla** [comment]
> test: let's add an e2e for this case

Consider it done :)

> @tmscarla issue: have you tried testing the frontend when served from the backend? Serving from the backend as a public folder differs from going to Next.js directly. If you try accessing a non-existing path on Flask at the moment you receive a 404 error, and the index.html is not served.

Yes I checked following the instructions you provided and I can confirm that it works! 👍🏻

**Review by mendaomn — DISMISSED** [review]

**Review by mtfranchetto — APPROVED** [review]

**Pull request merged**